cmd/scale.go

```diff
@@ -237,8 +239,31 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 				vmsToDelete = append(vmsToDelete, indexToVM[i])
 			}
 
-			if orchestratorInfo.OrchestratorType == api.Kubernetes {
-				err = sc.drainNodes(vmsToDelete)
+			switch orchestratorInfo.OrchestratorType {
+			case api.Kubernetes:
+				kubeConfig, err := acsengine.GenerateKubeConfig(sc.containerService.Properties, sc.location)
+				if err != nil {
+					log.Errorf("failed to generate kube config: %v", err)
+					return err
+				}
+				err = sc.drainNodes(kubeConfig, vmsToDelete)
+				if err != nil {
+					log.Errorf("Got error %+v, while draining the nodes to be deleted", err)
+					return err
+				}
+			case api.OpenShift:
+				bundle := bytes.NewReader(sc.containerService.Properties.OrchestratorProfile.OpenShiftConfig.ConfigBundles["master"])
+				fs, err := filesystem.NewTGZReader(bundle)
+				if err != nil {
+					log.Errorf("failed to read master bundle: %v", err)
+					return err
+				}
+				kubeConfig, err := fs.ReadFile("etc/origin/master/admin.kubeconfig")
```

Comment:
FWIW, In Kubernetes it seems they are generating a new kubeconfig so they are not reusing the one that was generated during cluster startup.

Reply:
Yes, but in the k8s case the actual authentication certificate within the kubeconfig is still the originally generated one.  While not perfect I think our approach is an improvement.